scripts/build_runtime_payload.py

```diff
@@ -0,0 +1,301 @@
+    if not os.path.exists(build_script):
+        raise Exception(f"Build script not found at path: {build_script}")
+
+    generate_layout_command = [build_script, "release", architecture, "generatelayoutonly"]
+
+    if not iswin():
+        generate_layout_command.extend(["-os", os_group])
+
+    if libraries_config:
+        generate_layout_command.append(f"/p:LibrariesConfiguration={libraries_config}")
+
+    RunCommand(generate_layout_command, verbose=True).run(runtime_repo_dir)
+
+    core_root_dir = os.path.join(
+        artifacts_tests_dir,
+        "coreclr",
+        f"{os_group}.{architecture}.Release",
+        "Tests",
+        "Core_Root",
+    )
+    if not os.path.exists(core_root_dir):
+        raise Exception(f"Core_Root directory not found in expected location: {core_root_dir}")
+
+    shutil.copytree(
+        core_root_dir,
+        core_root_dest,
+        dirs_exist_ok=True,
+        ignore=shutil.ignore_patterns("*.pdb"),  # Exclude PDBs (not needed in payloads)
+    )
+
+def build_mono_payload(
+    mono_payload_dst: str,
+    os_group: str,
+    framework: str,
+    build_config: str,
+    architecture: str,
+    product_version: str,
+    runtime_repo_dir: Optional[str] = None,
+    mono_archive_or_dir: Optional[str] = None,
+) -> None:
+    """Assemble a Mono testhost payload with corerun host files.
+
+    Copies testhost for the specified framework/OS/arch/config plus the `corerun`
+    host executable(s) from the coreclr layout to the shared framework directory
+    for the requested `product_version` (so BDN can launch with corerun).
+    """
+    if mono_archive_or_dir is None:
+        if runtime_repo_dir is None:
+            raise Exception("Please provide a path to the built mono artifacts")
+        mono_archive_or_dir = os.path.join(runtime_repo_dir, "artifacts", "bin")
+
+    build_config_upper = build_config.title()  # release -> Release
+
+    extract_archive_or_copy(
+        mono_archive_or_dir,
+        mono_payload_dst,
+        prefix=f"testhost/{framework}-{os_group}-{build_config_upper}-{architecture}/",
+    )
+    corerun_target = os.path.join(
+        mono_payload_dst, "shared", "Microsoft.NETCore.App", product_version
+    )
+    # No trailing slash: we only want files starting with "corerun" (e.g., corerun, corerun.exe)
+    extract_archive_or_copy(
+        mono_archive_or_dir,
+        corerun_target,
+        prefix=f"coreclr/{os_group}.{architecture}.{build_config_upper}/corerun",
+    )
+
+def build_monoaot_payload(
+    monoaot_artifacts_archive_or_dir: str, payload_dest: str, architecture: str
+) -> None:
+    """Build a Mono AOT payload consisting of cross tools + runtime pack.
+
+    The function expects either an artifacts directory or an archive containing
+    the structure produced by Mono AOT builds. Two extractions are performed:
+      1. Cross compiler / AOT toolchain for the given architecture.
+      2. The runtime pack published under `pack/` for consumption.
```

Comment:
This comment mentions that we are getting the runtime pack published under `pack/`, but the copy looks like it is not coming from the `pack` directory.

Reply:
Will update the comment to clarify, but more just meant that the runtime pack is put into a directory called pack, not from a directory called pack